debezium-api/src/main/java/io/debezium/engine/DebeziumEngine.java

```diff
@@ -176,6 +176,28 @@ default boolean supportsTombstoneEvents() {
         }
     }
 
+    interface Signaler<T> {
+
+        /**
+         * Initialize the signaler with the engine.
+         *
+         * <p>
+         *     Note that it is up the the implementation to decide whether it is compatible with given engine.
+         * </p>
+         *
+         * @param engine the engine instance
+         * @param <E> type of engine
+         */
+        <E extends DebeziumEngine<?>> void init(E engine);
```

Comment:
That's a possibility -- initialise with the list of tasks instead of the engine. However that brings us to the fact that on API layer there is no notion of task or anything. 

Or we could  actually initialise with the list of Signal channels, however then we will have to do few things

1) Bring signal channels (at least SignalChannelWriter) to the public API  (which they already sort of are as they are meant to be extendible anyway (nevertheless it would mean moving these contracts to debezium-api module)
2) Ensure that the write has information about the the specific tasks (or we could pass in something like `Map<task_id, signal_writer>` 

In all of these cases it means creating more detailed API. The current way essentially doesn't add anything to the API besides this interface (which doesn't expose any new information)

@jpechane  if you are up for it, I'm willing to expand the API. However in such case I would actually argue that there should also be a contract for the signal record.

Reply:
Maybe this is misunderstanding, I'm not definitely proposing extending the public API. I'm just proposing to change this one to `<E extends DebeziumEngine<?>> void init(E engine, String signalClassName);` or `<E extends DebeziumEngine<?>> void init(E engine, Class<S> signlaChannel);`

Reply:
The `String className` is imho not a good approach



```
<E extends DebeziumEngine<?>> void init(E engine, Class<S> signlaChannel);
```

Where does the S come from? The problem here is that this would require exposing the SignalChannelReader interface to public API... Actualy the SignalChannelWritter as that's the important thing here.